questionGenerator.py

```diff
@@ -185,6 +186,86 @@ def printTokenCount(self):
             f"Question Generation Token Count: {self.tokenCount}",
         )
 
+    def saveToFile(self):
+        """
+        Save the question data to a file.
+
+        Returns:
+            bool: True if the question data is successfully saved, False otherwise.
+        """
+        saveFolder = os.path.join(outputFolder, self.config.videoToUse)
+        try:
+            if not os.path.exists(saveFolder):
+                os.makedirs(saveFolder)
+        except OSError:
+            logging.warn(
+                f"Creation of the directory {saveFolder} failed. Data output will not be saved"
+            )
+            return False
+
+        questionSavePath = os.path.join(
+            outputFolder, self.config.videoToUse, "Questions.txt"
+        )
+        try:
+            with open(questionSavePath, "w") as f:
+                f.write(f"Video Name / Parent Folder: {self.config.videoToUse}\n")
+                for topic in self.responseData:
+                    response = self.responseData[topic]
+                    response = response.strip("` \n")
+                    if response.startswith("json"):
+                        response = response[4:]
+
+                    startTime = self.dominantTopics[topic]["Start"]
+                    endTime = self.dominantTopics[topic]["End"]
+                    durationMin, durationSec = divmod((endTime - startTime).total_seconds(), 60)
+
+                    f.write("\n---------------------------------------\n")
+                    f.write(f"Topic: {topic}\n")
+                    f.write(
+                        f"Transcipt Segment: {startTime.strftime('%H:%M:%S')}"
+                        + f" - {endTime.strftime('%H:%M:%S')}\n"
+                        + f"Duration: {int(durationMin)} minutes & {int(durationSec)} seconds\n"
+                    )
+                    f.write(
+                        f"Insert Point: {self.dominantTopics[topic]['End'].strftime('%H:%M:%S')}\n"
+                    )
+
+                    try:
```

Comment:
I would personally avoid Nested Try-catch.  You could just have the `JSONDecodeError` part of the Outer try-catch for readability. But again this is just an opinion and mostly people are split on this idea. 

https://mysteryweevil.medium.com/mastering-nested-try-except-blocks-in-python-a-comprehensive-guide-13f47679c130

Reply:
That's an interesting thing to think about. Let me read up more on this, thank you for the link.

Reply:
In this case, I will leave this as a nested try-catch, as the nested one also writes to the text file explicitly that something went wrong. This is because the JSON parsing can fail for some topics, but not all of them, so we would want the generated question file to still be readable regarding what topics have questions for them and what failed to be parsed correctly. The information for the question is still there, compared to the other try-catches where something failed and no data at all could be written.